### PR TITLE
Feat: dbt namespace macros

### PR DESCRIPTION
**Pull request by crericha**

### CODE REVIEW

File: sqlmesh/dbt/basemodel.py [hunk 1]
```diff
@@ -283,7 +283,7 @@ def __init__(self, context: DbtContext, config: BMC):
         self._rendered_sql: t.Optional[str] = None
         self._enriched_config: BMC = config.copy()
 
-        self._jinja_globals = create_builtin_globals(
+        self._jinja_globals = sqlmesh.dbt.builtin.create_builtin_globals(
```
Comment (izeigerman) [inline sqlmesh/dbt/basemodel.py:286]: why though?

File: sqlmesh/dbt/builtin.py [hunk 2]
```diff
@@ -341,6 +357,12 @@ def create_builtin_globals(
             }
         )
 
+    builtin_globals["dbt"] = (
+        _dbt_macros_registry()
+        .build_environment(**{**builtin_globals, **jinja_globals})
```
Comment (izeigerman) [inline sqlmesh/dbt/builtin.py:362]: seems like we do
```
{**builtin_globals, **jinja_globals}
```
twice. Should we do this instead:
```
builtin_globals.update(jinja_globals)

if "dbt" not in builtin_globals:
    builtin_globals["dbt"] = ...
    
return builtin_globals

File: sqlmesh/dbt/basemodel.py (hunk 1 above)
Comment (crericha) [reply sqlmesh/dbt/basemodel.py:286]: circular dependency

Comment (crericha) [reply sqlmesh/dbt/basemodel.py:286]: I don't have strong opinions on the topic. Do you prefer the local import?

Comment (tobymao) [reply sqlmesh/dbt/basemodel.py:286]: yea just do a local import

File: sqlmesh/dbt/builtin.py [hunk 3]
```diff
@@ -246,6 +250,18 @@ def _try_literal_eval(value: str) -> t.Any:
         return value
 
 
+def _dbt_macros_registry() -> JinjaMacroRegistry:
+    site_packages = next(p for p in sys.path if "site-packages" in p)
```
Comment (tobymao) [inline sqlmesh/dbt/builtin.py:254]: is this right? are you sure it's always the first one?

Comment (crericha) [reply sqlmesh/dbt/builtin.py:254]: As far as I can tell, yes. Would you prefer i look at all entries that include "site-packages"?

Comment (crericha) [reply sqlmesh/dbt/builtin.py:254]: Actually, I'll look for "site-packages/dbt".

**Review by izeigerman — APPROVED** [review]

**Pull request merged**